modules/apigee_edge_teams/src/Plugin/views/access/TeamPermission.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * Copyright 2020 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_teams\Plugin\views\access;
+
+use Drupal\apigee_edge_teams\Structure\TeamPermission as TeamPermissionEntity;
+use Drupal\apigee_edge_teams\TeamPermissionHandlerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\views\Plugin\views\access\AccessPluginBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Access plugin for team-permission-based access control.
+ *
+ * @ViewsAccess(
+ *   id = "team_permission",
+ *   title = @Translation("Team permission"),
+ * )
```

Comment:
This looks like a really great addition! However, I still don't see it getting used? (I might be getting ahead, I know the PR is still marked as draft).

Reply:
We are using it here (which somehow got lost during config update. I added it back).

https://github.com/apigee/apigee-edge-drupal/pull/474/files#diff-83c124c08fb13b9381165bcaded2220bR723-R726